exonum-java-binding-core/rust/src/utils/jni_cache.rs

```diff
@@ -23,7 +28,7 @@ static mut TRANSACTION_EXECUTION_EXCEPTION: Option<GlobalRef> = None;
 #[no_mangle]
 pub extern "system" fn JNI_OnLoad(vm: JavaVM, _: *mut c_void) -> jint {
     let env = vm.get_env().expect("Cannot get reference to the JNIEnv");
-    cache_methods(&env);
+    CACHE_INIT.call_once(|| cache_methods(&env));
```

Comment:
It is not a correct usage of once: https://doc.rust-lang.org/std/sync/struct.Once.html#examples-1

It won’t give you the memory guarantees that are needed in *reader* threads.